### PR TITLE
refactor!: deprecate report part of vim.health functions

### PR DESCRIPTION
**Pull request by dundargoc**

### CODE REVIEW

File: runtime/doc/news.txt [hunk 1]
```diff
@@ -68,6 +68,10 @@ The following changes may require adaptations in user config or plugins.
 
 • The default value of 'commentstring' is now empty instead of "/*%s*/".
 
+• Deprecate all functions with the "report_" prefix in the *vim.health*. For
```
Comment (clason) [inline runtime/doc/news.txt:71]: That should be in the "DEPRECATIONS" section at the end.

File: runtime/lua/vim/health.lua [hunk 2]
```diff
@@ -1,23 +1,48 @@
 local M = {}
 
+---@deprecated
 function M.report_start(msg)
-  vim.fn['health#report_start'](msg)
+  vim.deprecate('vim.health.report_start()', 'vim.health.start()', '0.10')
+  vim.fn['health#start'](msg)
```
Comment (clason) [inline runtime/lua/vim/health.lua:6]:
```suggestion
  M.start(msg)
```
and similarly below

File: runtime/autoload/health.vim [hunk 3]
```diff
@@ -60,6 +60,11 @@ function! s:collect_output(output)
   let s:output += split(a:output, "\n", 1)
 endfunction
 
+" Starts a new report.
+function! health#start(name) abort
+  call s:collect_output(printf("\n%s ~", a:name))
+endfunction
+
 " Starts a new report.
 function! health#report_start(name) abort
```
Comment (clason) [inline runtime/autoload/health.vim:69]: Shouldn't these be deprecated as well?

**Comment by clason** [comment]
+1 on the naming change, but let's turn this into a proper refactor/cleanup, including removing deprecated code :)

**Comment by dundargoc** [comment]
Superseded by https://github.com/neovim/neovim/pull/22927

**Pull request closed without merging**